AirCasting/Notes/EditNote/EditNoteViewModel.swift

```diff
@@ -14,19 +14,30 @@ class EditNoteViewModelDefault: EditNoteViewModel, ObservableObject {
     private var note: Note!
     private var notesHandler: NotesHandler
     private let exitRoute: () -> Void
+    private let sessionUpdateService: SessionUpdateService
     
     
-    init(exitRoute: @escaping () -> Void, noteNumber: Int, notesHandler: NotesHandler) {
+    init(exitRoute: @escaping () -> Void, noteNumber: Int, notesHandler: NotesHandler, sessionUpdateService: SessionUpdateService) {
         self.exitRoute = exitRoute
         self.notesHandler = notesHandler
+        self.sessionUpdateService = sessionUpdateService
         notesHandler.fetchSpecifiedNote(number: noteNumber) { note in
-            self.note = note
-            self.noteText = note.text
+            DispatchQueue.main.async {
+                self.note = note
+                self.noteText = note.text
+            }
         }
     }
     
     func saveTapped() {
-        notesHandler.updateNote(note: note, newText: noteText); exitRoute()
+        notesHandler.updateNote(note: note, newText: noteText, completion: {
```

Comment:
I feel like all this `sessionUpdateService` stuff should be handled inside `NotesHander` too. It's out use-case/controller object and it would make our lifes easier if it is 100% responsible for all bells and whistles of notes stuff.

We can add some methods to it if neccessary or change their names if they don't reflect anymore, but we shouldn't mix services into viewmodels (whenever we have proper controller object in place that is ofc)

Reply:
Sure, changing it. To be honest I was not sure where to put it back then. Thank you!